### PR TITLE
[release-0.44] Timeout Unschedulable Migration Target Pods

### PR DESCRIPTION
**Pull request by enp0s3**
Signed-off-by: Igor Bezukh <ibezukh@redhat.com>

This is a manual cherry-pick of #6737 

Fixes #
https://bugzilla.redhat.com/show_bug.cgi?id=2021992

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 1]
```diff
@@ -768,6 +772,20 @@ const (
 
 	// This annotation is to keep virt launcher container alive when an VMI encounters a failure for debugging purpose
 	KeepLauncherAfterFailureAnnotation string = "kubevirt.io/keep-launcher-alive-after-failure"
+
+	// MigrationTransportUnixAnnotation means that the VMI will be migrated using the unix URI
+	MigrationTransportUnixAnnotation string = "kubevirt.io/migrationTransportUnix"
```
Comment (jean-edouard) [inline staging/src/kubevirt.io/client-go/api/v1/types.go:777]: This doesn't seem to be part of the original PR, nor does it seem to be needed for the backported code.
Am I missing something?

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 2]
```diff
@@ -768,6 +772,20 @@ const (
...
+	// MigrationUnschedulablePodTimeoutSecondsAnnotation represents a custom timeout period used for unschedulable target pods
+	// This exists for functional testing
+	MigrationUnschedulablePodTimeoutSecondsAnnotation string = "kubevirt.io/migrationUnschedulablePodTimeoutSeconds"
+
+	// MigrationPendingPodTimeoutSecondsAnnotation represents a custom timeout period used for target pods stuck in pending for any reason
+	// This exists for functional testing
+	MigrationPendingPodTimeoutSecondsAnnotation string = "kubevirt.io/migrationPendingPodTimeoutSeconds"
+
+	// RealtimeLabel marks the node as capable of running realtime workloads
+	RealtimeLabel string = "kubevirt.io/realtime"
```
Comment (jean-edouard) [inline staging/src/kubevirt.io/client-go/api/v1/types.go:788]: Same as above

**Review by jean-edouard — COMMENTED** [review]
Couple comments in-line.
Note: while this technically passes the "API change" test, it just adds a few consts there, so it's probably fine to avoid backporting to all branches between 0.44 and main.

**Comment by enp0s3** [comment]
@jean-edouard Hi. Thanks for the review! Indeed the `RealtimeLabel` and `MigrationTransportUnixAnnotation` were unused, slipped during the merge conflict fix. Regarding the API change, the additional consts were for internal use for the e2e, I also think its not required to backport it all over till the main branch.

**Comment by enp0s3** [comment]
/hold
having some CI issues related to Bazel cache.

**Comment by enp0s3** [comment]
/close
was cherry-picked by mistake.

**Pull request closed without merging**